dagfactory/telemetry.py

```diff
@@ -45,15 +45,23 @@ def emit_usage_metrics(metrics: dict[str, object]) -> bool:
         **metrics, telemetry_version=constants.TELEMETRY_VERSION, query_string=query_string
     )
     logging.debug("Telemetry is enabled. Emitting the following usage metrics to %s: %s", telemetry_url, metrics)
-    response = httpx.get(telemetry_url, timeout=constants.TELEMETRY_TIMEOUT, follow_redirects=True)
-    if not response.is_success:
+    try:
+        response = httpx.get(telemetry_url, timeout=constants.TELEMETRY_TIMEOUT, follow_redirects=True)
+    except httpx.ConnectError as e:
```

Comment:
```suggestion
    except httpx.RequestError as e:
```

Similar to https://github.com/astronomer/astronomer-cosmos/pull/1439/commits/ae1eeeb72377ad25d579cec6868a296a26b36587

Reply:
shall we catch `httpx.HTTPError` to avoid task failure if any error occurs from the scarf server, too?

Reply:
Thanks for the suggestion, updated in 3ba0055cb109019d98f36aa188e8bd2332656193